pkg/ui/handlers.go

```diff
@@ -1,23 +1,32 @@
 package ui
 
 import (
+	"fmt"
 	"io/fs"
 	"net/http"
 	"path"
+	"regexp"
+	"strings"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	"github.com/go-chi/chi/v5"
 )
 
 type API struct {
 	fileServer http.Handler
-
-	logger logging.LoggerInterface
+	fileSystem fs.FS
+	logger     logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	// TODO @shipperizer unsure if we deal with any POST/PUT/PATCH via js
-	mux.HandleFunc("/*", a.uiFiles)
+	nodeList, _ := fs.ReadDir(a.fileSystem, ".")
+	r, _ := regexp.Compile(`html$`)
+	for _, node := range nodeList {
+		if r.MatchString(node.Name()) {
+			mux.HandleFunc(fmt.Sprintf("/%s", strings.Split(node.Name(), ".")[0]), a.uiFiles)
+		}
+	}
```

Comment:
just to keep the `RegisterEndpoints` lighter, split this in a separte private func

```
func (a *API) uiRoutes() []string {
    routes := make([]string, 0)
    
        nodeList, _ := fs.ReadDir(a.fileSystem, ".")
    	for _, node := range nodeList {
		if r.MatchString(node.Name()) {
			routes = append(routes, fmt.Sprintf("/%s", strings.Split(node.Name(), ".")[0]))
		}
	}
	return routes
}
```

or something similar